hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +60,115 @@
+        return mapConfigs.stream().map(MapConfig::getQueryCacheConfigs)
+                .filter(queryCacheConfigs -> !(queryCacheConfigs.isEmpty())).count();
     }
 
     private long countMapWithAtleastOneIndex() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getIndexConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getIndexConfigs)
+                .filter(indexConfigs -> !(indexConfigs.isEmpty())).count();
     }
 
     private long countMapWithHotRestartEnabled() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getHotRestartConfig().isEnabled()).count();
+        return mapConfigs.stream().map(MapConfig::getHotRestartConfig)
+                .filter(HotRestartConfig::isEnabled).count();
     }
 
     private long countMapWithWANReplication() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getWanReplicationRef() != null).count();
+        return mapConfigs.stream().map(MapConfig::getWanReplicationRef)
+                .filter(Objects::nonNull).count();
     }
 
     private long countMapWithAtleastOneAttribute() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getAttributeConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getAttributeConfigs)
+                .filter(attributeConfigs -> !(attributeConfigs.isEmpty())).count();
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static Predicate<MapConfig> isMapStoreEnabled() {
+        return p -> p.getMapStoreConfig().isEnabled();
+    }
+
+    static Predicate<MapConfig> isMapStoreDisabled() {
+        return p -> !p.getMapStoreConfig().isEnabled();
+    }
+
+    static class LatencyInfo {
+
+        private long totalLatency;
+        private long operationCount;
+
+        LatencyInfo(long totalLatency, long operationCount) {
+            this.totalLatency = totalLatency;
+            this.operationCount = operationCount;
+        }
+
+        void setLatency(long totalLatency, long totalOperation) {
```

Comment:
I think now there isn't much reason to keep this method around. The addition and assignment can be done in the `#add()` method.